File: Semester3/MAP/laboratoare/lab2/TaskManager/src/Domain/SortingTask.java
package Domain;

public class SortingTask extends Task {
    private int[] nrs;
    AbstractSorter sorter;

    public SortingTask(String taskId, String description, int[] nrs, AbstractSorter sorter) {
        super(taskId, description);
        this.nrs = nrs;
        this.sorter = sorter;
    }

    @Override
    public void execute() {
        sorter.sort(nrs);
    }
}

File: Semester3/MAP/Seminare/Sem10/Sem10/Task/SortingTask.cs
namespace Sem10;

public class SortingTask : Task
{
    private Sorter sorter;
    private int[] array;

    public SortingTask(string id, string description, int[] array, SortStrategy strategy) : base(id, description)
    {
        this.array = array;
        sorter = SorterFactory.getInstance().createSorter(strategy);
    }
    
    public override void execute()
    {
        sorter.sort(array);
        foreach (var nr in array)
        {
            Console.Write(nr + " ");
        }
        Console.WriteLine();
    }
}